### PR TITLE
Key Completion Review

### PR DESCRIPTION
**Pull request by AdamRoden**
@JuliusSweetland Here you can take a look at how I propose implementing this change. It's actually a nice step toward unity between built-in and dynamic keyboard capability. I suggest copying the current field into the new setting on application launch.

### CODE REVIEW

**Comment by JuliusSweetland** [comment]
I'll take a look as soon as possible, but I've got a full on week so please
bear with me.

File: src/JuliusSweetland.OptiKey.Core/OptiKeyApp.cs [hunk 1]
```diff
@@ -115,6 +115,18 @@ protected void Initialise()
                 Settings.Default.Reload();
             }
 
+            //Migrate the completion time setting to the new field
+            var timespan = Settings.Default.KeySelectionTriggerFixationDefaultCompleteTime;
+            if (timespan> TimeSpan.Zero)
+            {
+                Settings.Default.KeySelectionTriggerFixationDefaultCompleteTimes = timespan.TotalMilliseconds.ToString()
+                    + "," + Math.Round(0.75 * timespan.TotalMilliseconds).ToString()
```
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey.Core/OptiKeyApp.cs:123]: I'm aiming to look at this PR (and your other one) in the next few days. Can you clarify what this logic is doing?

Comment (AdamRoden) [reply src/JuliusSweetland.OptiKey.Core/OptiKeyApp.cs:123]: I can clarify the comment before finalizing the PR, but the intent here was to convert the old single value Completion Time into a list with the new format to nudge users toward exploring the feature. Another idea is to add conditional highlighting to the field in MC and show a message that repeat rates can be added.

**Comment by JuliusSweetland** [comment]
Ah I see. If this is for all key selection times then I wouldn't default
their old value to one which increases in selection speed. I fear some
users will not understand why keys are suddenly being selected multiple
times in faster succession. I think a mask on the textbox + a hint would go
a long way though.

**Comment by AdamRoden** [comment]
I added additional language to help users understand what they can enter, highlighted the field if it does not have multiple values, and added input validation.

**Comment by JuliusSweetland** [comment]
@AdamRoden Very nice PR. Merged. Thank you.

**Pull request merged**